source/Coop/Mod/CoopServer.cs

```diff
@@ -88,6 +95,8 @@ private void OnClientConnected(ConnectionServer connection)
             SyncedObjectStore.AddConnection(connection);
             connection.OnClientJoined += Persistence.ClientJoined;
             connection.OnDisconnected += Persistence.Disconnected;
+            connection.OnServerSendingWorldData += m_GameEnvironmentServer.LockTimeControlStopped;
+            connection.OnServerSendedWorldData += m_GameEnvironmentServer.UnlockTimeControl;
```

Comment:
I think that is too early to unlock. We should keep it paused until the client has loaded the state, since the synchronization will only start once the client has fully loaded into the game. I'd unlock on `connection.OnClientJoined` instead.